### DISCUSSION TITLE
Feature: Add backup/restore option for Docker containers

### DISCUSSION THREAD

**Discussion by Joooost** [IceWhaleTech/CasaOS#175] [Ideas]
#### Is your idea related to a problem? Please describe:
Backup and restore docker containers can be a challenge to setup and/or execute. This can be made easy by adding some features to easily backup and restore containers.

#### Describe the solution you'd like:
1. Add an option to backup all/selected Docker containers to a mounted or local directory.  The backup should contain everything to create/restore the complete docker container including appdata, volumes, network, ports and app details. Example archive:

docker_backup_20220510_150200.zip/
├─ HomeAssistant/
│  ├─ Appdata.zip
│  ├─ docker_compose.json
├─ FileBrowser/
│  ├─ Appdata.zip
│  ├─ docker_compose.json
├─ FileBrowser/
│  ├─ Appdata.zip
│  ├─ docker_compose.json
├─ backup.json (details of backup)

2. Add an option to schedule the backups (cronjob/setting)

3. Add an option to restore a backup by importing the backup archive file. On importing (via browser or path) a backup file the restore process should list the available Docker containers in the archive file and ask user which container(s) should be restored.  Restoring should be possible for the running CasaOS instance but also for a new and clean install.

#### Describe alternatives you've considered:
- Do it manually with shell scripts and tools which can be time consuming and challenging to setup.

#### Additional context
Additional requirements:
- When running backups the included containers should be stopped because of file and database integrity. 
- Nice to have is to optional password protect the archive file

Inspiration for scripts/tooling
- https://github.com/muesli/docker-backup
- https://github.com/piscue/docker-backup-scripts

**Comment by Tobias-Lauerbach** [comment]
An option to change the scheduled Backup to an Incremental Backup

**Reply by jsanders-morpheus** [reply]
The ability to have a backup of the links would be fantastic.  This would allow a quick rebuild or migration to different hardware.  If there was a way to autogen an install script of the containers installed that would be great as well, but I am sure that is a little more of an ask than just the link backup.

**Comment by reddexx** [comment]
Fulldd Backup for Backup complete System

**Comment by Yonny24** [comment]
Any progress on this? Any new apps to backup to google drive or locally? Perhaps a veeam agent app or something along those lines. Snapshots of the containers like we can do in proxmox?

**Comment by petruut** [comment]
would be great to have this implemented
